base/applications/mspaint/textedit.cpp

```diff
@@ -14,7 +14,7 @@
 
 /* FUNCTIONS ********************************************************/
 
-CTextEditWindow::CTextEditWindow() : m_hFont(NULL), m_hFontZoomed(NULL)
+CTextEditWindow::CTextEditWindow() : m_hFont(NULL), m_hFontZoomed(NULL), m_nMoveSizeLock(0)
```

Comment:
For what reason do we need a lock for that?

Reply:
```cxx
    InterlockedIncrement(&m_nMoveSizeLock);
    MoveWindow(rc.left, rc.top, rc.right - rc.left, rc.bottom - rc.top, FALSE);
    InterlockedDecrement(&m_nMoveSizeLock);
```
There was an error in positioning of the text box. This will fix that problem.

Reply:
Unless you have somehow two or more separate threads that would concurrently try to move the window, I have no idea why a single thread calling the moving would trigger a bug. Looks like a workaround to me.

Reply:
Retired to use `Interlocked...`.